### ISSUE TITLE
Request update of included renovate oss version in whitesource/renovate docker image

### ISSUE THREAD

**Issue by simon-weimann** [mend/renovate-on-prem#248]
We are using renovate on prem with our internal github and are quite happy with the provided service. However we noticed, that the last update of the docker image was almost 4 months ago (14.Oct) with the latest version of whitesource/renovate being `2.6.0` with renovate `32.238.1`

We are affected by the issue https://github.com/renovatebot/renovate/issues/18183 solved with https://github.com/renovatebot/renovate/pull/18402 that seems to be included from renovate `32.241.0` onwards.

Could you provide a new version of the whitesource/renovate docker image with an updated version of renovate oss?

Btw.: Do you have any kind of roadmap or schedule on when you will update the image? Seems like the updates got fewer over the last releases. Seems like whitesource/renovate is now already 2 major versions behind in terms of the included renovate oss version.

Thanks in advance!

**Comment by rarkins** [comment]
@simon-weimann @samsaggace I've pushed `whitesource/renovate:3.0.0-rc.0` just now: https://hub.docker.com/layers/whitesource/renovate/3.0.0-rc.0/images/sha256-7eaae68f877c06f6f85445c7fd5158f581ebb4303c472010535d9446add931ae?context=explore

It's a major release due to major release of Renovate (to v34 latest).

It's not yet tested and release notes not prepared, but I didn't want to delay it any further in case it's useful to you. I will try to test it and prepare the release notes in the next week.

**Comment by rarkins** [comment]
FYI also @jamietanna

**Comment by jamietanna** [comment]
Thanks Rhys, appreciate it 👏🏽 will try and get updated to use it later

**Comment by simon-weimann** [comment]
@rarkins Thank you for providing the release candidate. I have updated our test environment with the new `3.0.0-rc.0` version and it looks promising. 

I first had issues regarding the `RENOVATE_AUTODISCOVER_FILTER` environment variable regarding the helm chart and quoting. I had to enclose the variable in additional single quotes e.g. `"'cicd/**'".
I don't know if it is a side effect of https://github.com/renovatebot/renovate/issues/8763
Unfortunately the error message when starting up the server always lead me to an invalid cron syntax (which was not the case as I was using the default one).

Maybe you could improve the error message handling. Seems like any error on startup (e.g. when parsing the autodiscover filter or the cron syntax) always gets printed with the error message: 
```
INFO: Applying autodiscoverFilter
ERROR: Invalid cron value
```

Additionally it still seems that in `whitesource/renovate` only glob patterns are supported regarding the autodiscover filter. I couldn't get regex to work.

Besides that version `3.0.0-rc.0` works flawlessly for us. Looking forward to hearing from you about the new release in the upcoming week.

**Comment by jamietanna** [comment]
The only thing I think we've noticed is that Terraform versions aren't updated, with PRs created by the previous version of Renovate being autoclosed.

This is where

For instance, given code in our repository:

```hcl
terraform {
  required_version = "~> 1.1.0"
}
```

We would then get a Renovate PR with the following headers:

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| [hashicorp/terraform](https://togithub.com/hashicorp/terraform) | required_version | minor | `~> 1.1.0` -> `~> 1.3.0` |

---

### Release Notes

<details>
<summary>hashicorp/terraform</summary>

### [`v1.3.7`](https://togithub.com/hashicorp/terraform/releases/tag/v1.3.7)

[Compare Source](https://togithub.com/hashicorp/terraform/compare/v1.3.6...v1.3.7)

**Comment by rarkins** [comment]
Hashicorp versioning was refactored, so this may be a bug in Renovate OSS rather than anything specific to On-Prem

**Comment by jamietanna** [comment]
Gotcha, I'll raise an OSS issue for it. Wouldn't say it's a blocker for us, but you may disagree :)

**Comment by rarkins** [comment]
I have pushed 3.0.0-rc.1 which should provider better error logging for the filter/cron issue, but hold off using it until I can incorporate the fix in https://github.com/renovatebot/renovate/pull/19824

**Comment by kwinkel** [comment]
Regarding the repository filter: I've noticed that the string syntax is working, but the array one is not.
Using current onprem version v2.6.0.

Example 1: (working)
```js
module.exports = {
  "autodiscover": true,
  "autodiscoverFilter": ["group/sub1/project"]
}
```

```
 WARN: Config needs migrating
       "originalConfig": {"autodiscover": true, "autodiscoverFilter": ["group/sub1/project"]},
       "migratedConfig": {"autodiscover": true, "autodiscoverFilter": "group/sub1/project"}
```

```
TRACE: mergeChildConfig
       "parent": {"autodiscover": true, "autodiscoverFilter": "group/sub1/project"},
```

-> matches, looked up in log `Repository list from scheduler`

---

Example 2: (not working)
```js
module.exports = {
  "autodiscover": true,
  "autodiscoverFilter": [
    "group/sub1/project",
    "group/sub2/project"
  ]
}
```

```
TRACE: mergeChildConfig
       "parent": {
         "autodiscover": true,
         "autodiscoverFilter": [
           "group/sub1/project",
           "group/sub2/project"
         ]
       },
```

No line indiciating that the given config has been migrated.

-> matches zero times

Error:
```
ERROR: Unknown gitlabScheduler error
--
Mon, Jan 16 2023 9:40:01 pm | "err": {
Mon, Jan 16 2023 9:40:01 pm | "message": "invalid pattern",
Mon, Jan 16 2023 9:40:01 pm | "stack": "TypeError: invalid pattern\n at assertValidPattern (/usr/src/app/node_modules/minimatch/minimatch.js:275:11)\n at minimatch (/usr/src/app/node_modules/minimatch/minimatch.js:116:3)\n at /usr/src/app/src/server.js:1:13998\n at Array.filter (<anonymous>)\n at gitlabScheduler (/usr/src/app/src/server.js:1:13988)\n at processTicksAndRejections (node:internal/process/task_queues:96:5)"
```

---

Example 3: (not working)
```js
module.exports = {
  "autodiscover": true,
  "autodiscoverFilter": "group/sub1/project, group/sub2/project"
}
```

```
TRACE: mergeChildConfig
       "parent": {
         "autodiscover": true,
         "autodiscoverFilter": "group/sub1/project, group/sub2/project"
       },
```

No line indiciating that the given config has been migrated.

-> matches zero times

**Comment by rarkins** [comment]
Is the behavior the same when using the Renovate OSS CLI?

**Comment by kwinkel** [comment]
With Renovate OSS CLI, it did work like described in the documentation.
https://docs.renovatebot.com/self-hosted-configuration/#autodiscoverfilter

I've used a filter with a list of project paths there.

**Comment by samsaggace** [comment]
Hello, 
Do you have some news on an official release of the v3 ? 🙏 
Sadly I don't have a test environment where I can check the behaviour without impacting all my repositories...

Thanks again

**Comment by rarkins** [comment]
Now released 3.0.0 officially

**Comment by samsaggace** [comment]
Thanks a lot, that's awesome ! 🙏
